common/Environments/Helpers/ComputeSystemHelpers.cs

```diff
@@ -126,4 +129,30 @@ public static EnvironmentsCallToActionData UpdateCallToActionText(int providerCo
 
         return new(navigateToExtensionsLibrary, callToActionText, callToActionHyperLinkText);
     }
+
+    /// <summary>
+    /// Safely remove all items from an observable collection.
+    /// </summary>
+    /// <remarks>
+    /// There can be random COM exceptions due to using the "Clear()" method in an observable collection. This method
+    /// is used so that we can safely clear the observable collection without throwing an exceptions. This is related
+    /// to this GitHub issue https://github.com/microsoft/microsoft-ui-xaml/issues/8684. To work around this,
+    /// this method is used to remove all items individually from the end of the collection to the beginning of the collection.
+    /// </remarks>
+    /// <typeparam name="T">Type of objects that the collection contains</typeparam>
+    /// <param name="collection">An observable collection that contains zero to N elements</param>
+    public static void RemoveAllItems<T>(ObservableCollection<T> collection)
+    {
+        try
+        {
+            for (var i = collection.Count - 1; i >= 0; i--)
+            {
+                collection.RemoveAt(i);
+            }
+        }
+        catch (Exception ex)

```

Comment:
If we get an exception, would we still want to try to clear the rest of the collection, or is it not worth it?

Reply:
I believe it's not worth it, as it may result in another exception

Reply:
It would be nice to figure out the root cause of the issue. May be open a lower priority bug. Then we could understand what's the best workaround is. For example, if it's caused by a race then using a lock around Clear() could be a better solution.
One side effect of removing items one by one could be sending too many/often notifications to the subscribers, which can affect UX responsiveness or cause some issues within the subscribers if they didn't expect that.